### DISCUSSION TITLE
TypeError: Cannot read property 'useRef' of null

### DISCUSSION THREAD

**Discussion by Jscripter-pk** [TanStack/query#4842] [Q&A]
I created an SDK which will use react-query bindings so that it can be used in **React** and **ReactNative** app.

I am export this from a client class

```

useLoanHooks() {
    return {
      loanData: (loansQueryDto: LoansQueryDto) => {
        return useQuery({
          queryKey: 'loanData',
          queryFn: async () => await this.loanData(loansQueryDto),
          keepPreviousData: true,
        });
      },
  }

  ```
  
  And test for this in SDK is below which works for me
  
  ```
  it('gets list of the loans available', async () => {
      const { loanData } = client.useLoanHooks();
      const { result } = renderHook(() => loanData({
        offset: '5',
        limit: '3',
      }), {
        wrapper,
      });
      await act(async () => {
        await result.current.refetch();
      });
      await waitFor(() => expect(result.current.isSuccess).toBe(true));
      expect(result.current.data).toBeDefined();
    }, 30000);
  
  ```
  
   And then for using doing it in mobile I do like this on which I get the error 
    **TypeError: Cannot read property 'useRef' of null**
    
  ```
  
  const loanHook = krossClient.loans.useLoanHooks();
  const {data, isFetching} = loanHook.loanData({
    offset: '0',
  });
  
  ```
  
  @TkDodo Can I get any clue why this might be happening. I might be doing something wrong
  Note: When I normally use the function with useQuery from react-native I don't get this but when I try to use it from SDK I get the error

**Comment by ilteoood** [comment]
Sorry to resurrect this, but I'm actually facing the same issue.
Have you been able to find a solution @Jscripter-pk?
